jujugui/static/gui/src/app/components/header-logo/header-logo.js

```diff
@@ -27,15 +27,43 @@ YUI.add('header-logo', function() {
   juju.components.HeaderLogo = React.createClass({
 
     propTypes: {
+      gisf: React.PropTypes.bool.isRequired,
+      showProfile: React.PropTypes.func
+    },
+
+    /**
+      Calls the showProfile prop
+      @param {Object} e The click event.
+    */
+    _showProfile: function(e) {
+      e.preventDefault();
+      this.props.showProfile();
+    },
+
+    /**
+      Returns the Juju logo SvgIcon component.
+      @return {Object} The SvgIcon component of the Juju logo.
+    */
+    _svg: function() {
+      return (<juju.components.SvgIcon name="juju-logo"
+        className="svg-icon"
+        width="90" height="35" />);
     },
 
     render: function() {
+      if (this.props.gisf) {
+        return (
+          <a href="/"
+             role="button"
+             title="Home">
+            {this._svg()}
+          </a>);
+      }
       return (
-        <a href="/"
-          role="button" title="Go to user profile">
-          <juju.components.SvgIcon name="juju-logo"
-            className="svg-icon"
-            width="90" height="35" />
+        <a onClick={this._showProfile}
```

Comment:
Use `generatePath`'s new ability here to add an `href` attribute pointing to the profile.